src/main/java/ladder/domain/Line.java

```diff
@@ -0,0 +1,45 @@
+package ladder.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import ladder.exception.PersonCountException;
+import ladder.strategy.NoMovableStrategy;
+import ladder.strategy.RandomlyMovableStrategy;
+
+public class Line {
+    private static final String PERSON_MIN_NUMBER_EXCEPTION_COMMENT = "사람 수는 최소 %d명 이상이어야 합니다.";
+    private static final int PERSON_MIN_NUMBER = 2;
+    private static final int START_INDEX_MIDDLE_LINE = 1;
+
+    private final List<Point> points;
+
+    private Line(int countOfPerson) {
+        validate(countOfPerson);
+        List<Point> points = new ArrayList<>();
+        constructLine(countOfPerson, points);
+        this.points = Collections.unmodifiableList(points);
+    }
+
+    public static Line from(int countOfPerson) {
+        return new Line(countOfPerson);
+    }
```

Comment:
```suggestion
    private Line(List<Point> points) {
        this.points = points;
    }

    public static Line from(List<Point> points) {
        return new Line(points);
    }

    public static Line from(int countOfPerson) {
        validate(countOfPerson);
        return from(constructLine(countOfPerson));
    }
```
이렇게 표현할 수 있도록 만들어주면 어떨까요?
`Line`을 여러 형태의 input으로 만들 수 있도록 했는데 이걸 객체지향에서 `다형성` 이라고 한답니다 ㅎㅎ

Reply:
다형성을 머리로만 이해하고 사용할 줄 아는 것은 다른 것 같습니다.
감사합니다.